src/distrs/pois.jl

```diff
@@ -12,19 +12,8 @@ import .RFunctions:
     poisinvlogcdf,
     poisinvlogccdf
 
-# generic versions
+# Julia implementations
 poispdf(λ::Real, x::Real) = exp(poislogpdf(λ, x))
 
+poislogpdf(λ::Real, x::Real) = poislogpdf(promote(λ, x)...)
 poislogpdf(λ::T, x::T) where {T <: Real} = xlogy(x, λ) - λ - loggamma(x + 1)
-
-poislogpdf(λ::Number, x::Number) = poislogpdf(promote(float(λ), x)...)
```

Comment:
Do you know why `float` was used previously?

Reply:
No.

Reply:
Maybe to ensure that `poislogpdf(::Rational{Int}, ::Int)` also hits the R fallback?